src/staff_ghosts.c

```diff
@@ -212,261 +212,275 @@ void func_80005310(void) {
+    sButtonsPrevReplay = buttons;
+    gControllerEight->button = buttons;
 
-    if (D_80162DAC == 0) {
-        D_80162DB0++;
-        D_80162DAC = (s32) (D_80162DB4[D_80162DB0] & 0xFF0000);
+    if (sReplayFramesRemaining == 0) {
+        sReplayDataIdx++;
+        sReplayFramesRemaining = (s32) (replayData[sReplayDataIdx] & 0xFF0000);
     } else {
-        D_80162DAC += 0xFFFF0000;
+        sReplayFramesRemaining -= 0x10000;
     }
 }
 
-void func_8000561C(void) {
-    u32 temp_a0;
-    u32 temp_v0;
+void process_course_ghost_inputs(void) {
+    u32 inputs;
+    u32 stickBytes;
     UNUSED u16 unk;
-    u16 temp_v1;
-    s16 phi_v1;
-    s16 phi_a2 = 0;
-
-    if (D_80162DA0 >= 0x1000) {
+    u16 buttonsTemp;
+    s16 stickVal;
+    s16 buttons = 0;
+    if (sCourseGhostDataIndex >= 0x1000) {
         func_80005AE8(gPlayerThree);
         return;
     }
-    temp_a0 = D_80162DA4[D_80162DA0];
-    temp_v0 = temp_a0 & 0xFF;
-    if (temp_v0 < 0x80U) {
-        phi_v1 = (s16) (temp_v0 & 0xFF);
-    } else {
-        phi_v1 = (s16) (temp_v0 | (~0xFF));
+    
+    inputs = sCourseGhostData[sCourseGhostDataIndex];
+    stickBytes = inputs & 0xFF;
+    // converting signed 8-bit values to signed 16-bit values
+    if (stickBytes < 0x80U) {
+        stickVal = (s16) (stickBytes & 0xFF);
     }
+    else {
+        stickVal = (s16) (stickBytes | (~0xFF));
+    }
+    stickBytes = (u32) (inputs & 0xFF00) >> 8;
+    gControllerSeven->rawStickX = stickVal;
 
-    temp_v0 = (u32) (temp_a0 & 0xFF00) >> 8;
-    gControllerSeven->rawStickX = phi_v1;
-
-    if (temp_v0 < 0x80U) {
-        phi_v1 = (s16) (temp_v0 & 0xFF);
+    if (stickBytes < 0x80U) {
+        stickVal = (s16) (stickBytes & 0xFF);
     } else {
-        phi_v1 = (s16) (temp_v0 | (~0xFF));
+        stickVal = (s16) (stickBytes | (~0xFF));
     }
-    gControllerSeven->rawStickY = phi_v1;
+    gControllerSeven->rawStickY = stickVal;
 
-    if (temp_a0 & 0x80000000) {
-        phi_a2 = A_BUTTON;
+    if (inputs & 1 << 31) {
+        buttons = A_BUTTON;
     }
-    if (temp_a0 & 0x40000000) {
-        phi_a2 |= B_BUTTON;
+    if (inputs & 1 << 30) {
+        buttons |= B_BUTTON;
     }
-    if (temp_a0 & 0x20000000) {
-        phi_a2 |= Z_TRIG;
+    if (inputs & 1 << 29) {
+        buttons |= Z_TRIG;
     }
-    if (temp_a0 & 0x10000000) {
-        phi_a2 |= R_TRIG;
+    if (inputs & 1 << 28) {
+        buttons |= R_TRIG;
     }
-
-    temp_v1 = gControllerSeven->buttonPressed & 0x1F0F;
-    gControllerSeven->buttonPressed = (phi_a2 & (phi_a2 ^ D_80162D98)) | temp_v1;
-    temp_v1 = gControllerSeven->buttonDepressed & 0x1F0F;
-    gControllerSeven->buttonDepressed = (D_80162D98 & (phi_a2 ^ D_80162D98)) | temp_v1;
-    D_80162D98 = phi_a2;
-    gControllerSeven->button = phi_a2;
-    if (D_80162D9C == 0) {
-        D_80162DA0++;
-        D_80162D9C = (s32) (D_80162DA4[D_80162DA0] & 0xFF0000);
+    // Blanks the A, B, Z, R and L buttons
+    buttonsTemp = gControllerSeven->buttonPressed & 0x1F0F;
```

Comment:
Comment here is good! Please put a define for 0x1F0F here as well and below here as well.